### PR TITLE
Clarify online softmax split reduction limitation and invite contributions (refs #153241)

### LINKED ISSUE #153241: Online softmax is disabled on the fly
### 🐛 Describe the bug

Hi there, I started getting this warning when running a `torch.compile()`'d transformer forward inference pass with `mode="reduce-overhead", fullgraph=True`

```
/app/venv/lib/python3.12/site-packages/torch/_inductor/lowering.py:7007: UserWarning: 
Online softmax is disabled on the fly since Inductor decides to
split the reduction. Cut an issue to PyTorch if this is an
important use case and you want to speed it up with online
softmax.
```

I noticed this appears to be a TODO? https://github.com/pytorch/pytorch/blame/main/torch/_inductor/lowering.py#L7058C50-L7058C50

So I guess to whom it may concern, perhaps @shunting314, it does appear that inference needs split online softmax.

### Versions

Collecting environment information...
PyTorch version: 2.7.0+cu128
Is debug build: False
CUDA used to build PyTorch: 12.8
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.1 LTS (aarch64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0
Clang version: Could not collect
CMake version: version 3.28.3
Libc version: glibc-2.39

Python version: 3.12.3 (main, Feb  4 2025, 14:48:35) [GCC 13.3.0] (64-bit runtime)
Python platform: Linux-6.8.0-1013-nvidia-64k-aarch64-with-glibc2.39
Is CUDA available: True
CUDA runtime version: 12.8.93
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GH200 480GB
Nvidia driver version: 570.86.16
cuDNN version: Probably one of the following:
/usr/lib/aarch64-linux-gnu/libcudnn.so.9.8.0
/usr/lib/aarch64-linux-gnu/libcudnn_adv.so.9.8.0
/usr/lib/aarch64-linux-gnu/libcudnn_cnn.so.9.8.0
/usr/lib/aarch64-linux-gnu/libcudnn_engines_precompiled.so.9.8.0
/usr/lib/aarch64-linux-gnu/libcudnn_engines_runtime_compiled.so.9.8.0
/usr/lib/aarch64-linux-gnu/libcudnn_graph.so.9.8.0
/usr/lib/aarch64-linux-gnu/libcudnn_heuristic.so.9.8.0
/usr/lib/aarch64-linux-gnu/libcudnn_ops.so.9.8.0
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                         aarch64
CPU op-mode(s):                       64-bit
Byte Order:                           Little Endian
CPU(s):                               64
On-line CPU(s) list:                  0-63
Vendor ID:                            ARM
Model name:                           Neoverse-V2
Model:                                0
Thread(s) per core:                   1
Core(s) per cluster:                  64
Socket(s):                            -
Cluster(s):                           1
Stepping:                             r0p0
BogoMIPS:                             2000.00
Flags:                                fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 sm3 sm4 asimddp sha512 sve asimdfhm dit uscat ilrcpc flagm ssbs sb paca pacg dcpodp sve2 sveaes svepmull svebitperm svesha3 svesm4 flagm2 frint svei8mm svebf16 i8mm bf16 dgh bti
NUMA node(s):                         9
NUMA node0 CPU(s):                    0-63
NUMA node1 CPU(s):                    
NUMA node2 CPU(s):                    
NUMA node3 CPU(s):                    
NUMA node4 CPU(s):                    
NUMA node5 CPU(s):                    
NUMA node6 CPU(s):                    
NUMA node7 CPU(s):                    
NUMA node8 CPU(s):                    
Vulnerability Gather data sampling:   Not affected
Vulnerability Itlb multihit:          Not affected
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Not affected
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Not affected
Vulnerability Spec rstack overflow:   Not affected
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:             Mitigation; __user pointer sanitization
Vulnerability Spectre v2:             Not affected
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Not affected

Versions of relevant libraries:
[pip3] numpy==2.2.5
[pip3] torch==2.7.0+cu128
[pip3] torchao==0.10.0
[pip3] triton==3.3.0
[conda] Could not collect

### PR DESCRIPTION
**Pull request by ArjunJagdale**
## Summary

This PR updates the `prepare_softmax_online` lowering in `torch/_inductor/lowering.py` to clarify the current limitation: split reductions for online softmax are not supported. The warning message and comments are improved to provide more actionable information to users and contributors, including a direct link to the tracking issue (#153241).

## Motivation

- Make it clear to users why online softmax may be disabled at runtime and how they can help prioritize the feature by engaging with the tracking issue.
- Encourage contributions by explicitly marking the spot as a good place to help, with a relevant TODO and link to the discussion.
- Help future maintainers and users understand the context and current state of this limitation.

## Changes

- Updated the docstring for `prepare_softmax_online` to document the split reduction limitation and link to the tracking issue.
- Improved the inline comments to clarify the technical challenge and to welcome contributions.
- Enhanced the warning message to be more actionable, with a link to the relevant GitHub issue.

## Related Issue

Closes #153241

## Additional Notes

If you have an inference workload that would benefit from online softmax split reduction, please join the discussion on #153241. Contributions are welcome!

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

### CODE REVIEW

**Review by eellison — COMMENTED** [review]
Deferring to shunting. but, generally i would say please no refactor comment pr since they are trivial changes.
